src/Aspire.Dashboard/Otlp/Model/ColorGenerator.cs

```diff
@@ -2,64 +2,65 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Collections.Concurrent;
-using System.Globalization;
 
 namespace Aspire.Dashboard.Otlp.Model;
 
-public sealed class GeneratedColor
+public sealed class AccentColor
 {
-    public required string Hex { get; init; }
-    public required int Red { get; init; }
-    public required int Green { get; init; }
-    public required int Blue { get; init; }
+    public AccentColor(string variableName)
+    {
+        VariableName = variableName;
+        ReferencedVariableName = $"var({variableName})";
+    }
+
+    public string VariableName { get; }
+    public string ReferencedVariableName { get; }
 }
 
+/// <summary>
+/// Provides a stable color for a named element. When <see cref="GetColorVariableByKey(string)" /> is invoked a new color is returned if the key was not used previously. An instance of this class is thread-safe and multiple threads can query colors concurrently without collisions.
+/// </summary>
 public class ColorGenerator
 {
-    private static readonly string[] s_colorsHex =
+    private static readonly string[] s_variableNames =
     [
-        "#17B8BE", "#F8DCA1", "#B7885E", "#FFCB99", "#F89570",
-        "#829AE3", "#E79FD5", "#1E96BE", "#89DAC1", "#B3AD9E",
-        "#12939A", "#DDB27C", "#88572C", "#FF9833", "#EF5D28",
-        "#162A65", "#DA70BF", "#125C77", "#4DC19C", "#776E57"
+        "--accent-teal",
+        "--accent-marigold",
+        "--accent-brass",
+        "--accent-peach",
+        "--accent-coral",
+        "--accent-royal-blue",
+        "--accent-orchid",
+        "--accent-brand-blue",
+        "--accent-seafoam",
+        "--accent-mink",
+        "--accent-cyan",
+        "--accent-gold",
+        "--accent-bronze",
+        "--accent-orange",
+        "--accent-rust",
+        "--accent-navy",
+        "--accent-berry",
+        "--accent-ocean",
+        "--accent-jade",
+        "--accent-olive"
     ];
     public static readonly ColorGenerator Instance = new ColorGenerator();
 
-    private readonly List<GeneratedColor> _colors;
+    private readonly List<AccentColor> _colors;
```

Comment:
FYI another option would have been to use the hash of the key like

```c#
return _colors[Math.Abs(i.GetHashCode()) % colors.Length];
```

That prevents the dictionary, the lazy, concurrency management.
It's not as optimized since two resources may get the same color but the odds are still low. Commonly used to fit values pseudo-randomly in a specific number of buckets.